server/sonar-process/src/main/java/org/sonar/process/systeminfo/ProcessStateProvider.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * SonarQube
+ * Copyright (C) 2009-2016 SonarSource SA
+ * mailto:contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.process.systeminfo;
+
+import java.lang.management.ManagementFactory;
+import java.lang.management.MemoryMXBean;
+import java.lang.management.MemoryUsage;
+import java.lang.management.ThreadMXBean;
+import javax.annotation.CheckForNull;
+import org.sonar.process.systeminfo.protobuf.ProtobufSystemInfo;
+
+public class ProcessStateProvider implements SystemInfoSectionProvider {
+  private static final long MEGABYTE = 1024L * 1024L;
+  private final String name;
+
+  public ProcessStateProvider(String name) {
+    this.name = name;
+  }
+
+  @Override
+  public ProtobufSystemInfo.Section toSystemInfoSection() {
+    return toSystemInfoSection(ManagementFactory.getMemoryMXBean());
+  }
+
+  // Visible for testing
+  ProtobufSystemInfo.Section toSystemInfoSection(MemoryMXBean memoryBean) {
```

Comment:
I would have named this method `toProtobuf`, I don't see the value in the override of method `toSystemInfoSection` and I find the method name does not describe what it is doing